feat: M -> D native discord edits

Vyquos opened the pull request:
fixes issue #644.

Vyquos commented:
A few things to consider:
1. the edits aren't stored in `DiscordBot.store`
2. the db reference of the old matrix event and discord message isn't updated
3. the old mechanism of deleting and re-sending the edited discord message is kept as a fallback - though I'm not sure that it's necessary

Miepee commented:
Not a reviewer, but what was the reason for bumping the better-discord.js dependency?

Vyquos commented:
With [48b0430](https://github.com/Sorunome/better-discord.js/commit/48b0430770f453be4533d044e63bf2d9aebe112d "commit 48b0430 on better-discord.js"), the `discord.Webhook.editMessage` endpoint is added. Because `better-discord.js` is very outdated (last commit was made >1.5 years ago) i thought I'd bump the version of the dependency to its most recent.
There's only one commit more recent than the one i need (48b0430) anyway.

Vyquos committed:
Change misleading variable name

Rename "editEventId" to "editedEventId" in DiscordBot.edit. The id string that's
handed to the function as an argument is the id of the original message event,
while 'event.event_id' is the actual replace/edit-event.

Vyquos committed:
Bridge edits to discord webhooks

Support M->D edit bridging via the native discord webhook endpoint

Fixes issue #644

Vyquos committed:
Add changelog

Miepee commented on changelog.d/819.feature:
```diff
@@ -0,0 +1 @@
+Bridge edits on matrix to native discord edits instead of deleting and resending.
```
```suggestion
Bridge edits on Matrix to native Discord edits, instead of deleting and resending messages.
```

Vyquos committed:
Fix changelog typo

Co-authored-by: Miepee <38186597+Miepee@users.noreply.github.com>
Signed-off-by: Vyquos <75266055+Vyquos@users.noreply.github.com>

Vyquos commented:
thanks!

Miepee commented:
With the merge of #858 you probably need to rebase this @Vyquos

Vyquos commented:
thanks for the reminder! My private signoff is still valid, right?

Miepee commented:
Am not a maintainer, just some guy trying to help here. Someone from the orga needs to answer that.

fee1-dead commented:
Any updates on this?

Vyquos commented:
<s>Accidentally closed this while rebasing. The issue remains. I can't reopen it so I'll open a new one, but I'd be happy to continue working on this PR if the maintainers could reopen this issue.</s>

I just saw that I can just reopen the issue. Oops, my bad.

Vyquos commented:
> Any updates on this?

I rebased again, hoping for review!
